waterbutler/providers/rushfiles/provider.py

```diff
@@ -134,9 +149,31 @@ async def upload(self,
 
     async def delete(self,  # type: ignore
                      path: RushFilesPath,
-                     confirm_delete: int=0,
                      **kwargs) -> None:
-        raise NotImplementedError
+        if not path.identifier:
+            raise exceptions.NotFoundError(str(path))
+        if path.is_root:
+            raise exceptions.DeleteError(
+                'root cannot be deleted',
+                code=400
+            )
+
+        response = await self.make_request(
+                        'DELETE',
+                        self._build_filecache_url(str(self.share['id']), 'files', path.identifier),
+                        data=json.dumps({
+                            "TransmitId": self._generate_uuid(),
+                            "ClientJournalEventType": 1,
+                            "DeviceId": "waterbutler "
+                        }),
+                        headers={'Content-Type': 'application/json'},
+                        expects=(200, 404,),
+                        throws=exceptions.DeleteError,
+                    )
+        if response.status == 404:
```

Comment:
404 is returned when the share doesn't exist. For files, filecache returns 400. I think we should handle both or neither. Not only one.

```
HTTP/1.1 400 Bad Request
Content-Type: application/json; charset=utf-8
Server: Microsoft-IIS/10.0
X-Powered-By: ASP.NET
Access-Control-Allow-Origin: *
Access-Control-Allow-Methods: GET, POST, PUT, DELETE
Date: Wed, 24 Nov 2021 20:56:44 GMT
Content-Length: 188

{
  "Code": 17,
  "Data": {},
  "Message": "The file with the given internalname does not exists.",
  "ResponseInfo": {
    "ResponseCode": 1022,
    "Reference": "135892172"
  }
}
```